libs/platform/ebpf_pinning_table.h

```diff
@@ -12,6 +12,15 @@ extern "C"
 #endif
     typedef struct _ebpf_pinning_table ebpf_pinning_table_t;
 
+    /**
+     * @brief eBPF pinning table entry.
+     */
+    typedef struct _ebpf_pinning_entry
+    {
+        ebpf_utf8_string_t name;
+        ebpf_object_t* object;
+    } ebpf_pinning_entry_t;
```

Comment:
This is supposed to be an implementation detail of this module. Why are we exposing it outside the module? If we need to update the internals of this module, we now need to touch everyone who uses it.

Reply:
This is because I didn't want pinning_table to be aware of object types it contains. That is, I did not want to write a function that enumerated maps. So I had to return objects and their names. Since this struct was already there, I exposed it via the header. 2 Options:
1) Keep this struct exposed. Any additional changes in pinning_table should declare a new struct and "derive" from this struct.
2) Make this opaque, and expose a couple get() functions.
Preference?

Reply:
I am inclined towards the get function approach.

Reply:
Sounds good to me

Reply:
that's a great point.  I'm fine with making ebpf_pinning_entry_t be "public".